ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraph/ColorStackLegend.tsx

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import cx from 'classnames';
+import {Icon} from '@iconify/react';
+import useUserPreference, {USER_PREFERENCES} from '@parca/functions/useUserPreference';
+import {EVERYTHING_ELSE, selectStackColors, useAppSelector} from '@parca/store';
+import {useMemo} from 'react';
+import {useURLState} from '@parca/components';
+import type {NavigateFunction} from '@parca/functions';
+
+interface Props {
+  navigateTo?: NavigateFunction;
+  compareMode?: boolean;
+}
+
+const ColorStackLegend = ({navigateTo, compareMode = false}: Props): JSX.Element => {
```

Comment:
There seems to be some inconsistencies between the spacing of the color legends, and also the `x` icon gets shrunk in some cases.

<img width="735" alt="image" src="https://user-images.githubusercontent.com/9016992/215802740-fc4f0b77-a27d-4b65-afb5-30a4b92dde81.png">

<img width="788" alt="image" src="https://user-images.githubusercontent.com/9016992/215802850-965228d9-ee9a-4d93-b111-fcebba487f17.png">

We could also try out making the legends start from the beginning of the div.
<img width="663" alt="image" src="https://user-images.githubusercontent.com/9016992/215804948-495d7d0c-9c90-4a04-be1a-3e2ebf151474.png">

Reply:
> There seems to be some inconsistencies between the spacing of the color legends, and also the `x` icon gets shrunk in some cases.

Was using `grid` that made the spacing look uneven, now changed it to `flex` and it looks better.

> We could also try out making the legends start from the beginning of the div.

Fixed this too.